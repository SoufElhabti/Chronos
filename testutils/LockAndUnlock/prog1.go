package pkg

import (
	"sync"
)

var a int

func fn1() {
	mutex := sync.Mutex{}
	mutex.Lock()
	mutex.Unlock()
	a = 5
}
