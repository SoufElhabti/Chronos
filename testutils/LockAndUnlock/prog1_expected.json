{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {
			"t01598534": 1598583
		}
	},
	"GuardedAccess": [
		{
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				},
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {
						"t01598534": 1598583
					}
				}
			}
		},
		{
			"Value": 1598512,
			"OpKind": 1,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				},
				"LocksetJson": {
					"ExistingLocks": {},
					"ExistingUnlocks": {
						"t01598534": 1598583
					}
				}
			}
		}
	]
}